### PR TITLE
[Feature][Connector]add s3file save mode function

### PR DESCRIPTION
**Pull request by chl-wxp**
### Check list

* [x] add save mode function for s3file sink

### CODE REVIEW

File: docs/en/connector-v2/sink/S3File.md [hunk 1]
```diff
@@ -241,6 +246,22 @@ When File Format is Excel,The maximum number of data items that can be cached in
 
 Writer the sheet of the workbook
 
+### schema_save_mode[Enum]
+
+Before turning on the synchronous task, do different treatment of the target path.  
+Option introduction：  
+`RECREATE_SCHEMA` ：Will create when the path does not exist, delete and rebuild when the path is saved
```
Comment (EricJoy2048) [inline docs/en/connector-v2/sink/S3File.md]:
```suggestion
`RECREATE_SCHEMA` ：Will be created when the path does not exist. If the path already exists, delete the path and recreate it.
```

File: docs/en/connector-v2/sink/S3File.md [hunk 2]
```diff
@@ -241,6 +246,22 @@ When File Format is Excel,The maximum number of data items that can be cached in
 
 Writer the sheet of the workbook
 
+### schema_save_mode[Enum]
+
+Before turning on the synchronous task, do different treatment of the target path.  
+Option introduction：  
+`RECREATE_SCHEMA` ：Will create when the path does not exist, delete and rebuild when the path is saved        
+`CREATE_SCHEMA_WHEN_NOT_EXIST` ：Will Created when the path does not exist, skipped when the path is saved
```
Comment (EricJoy2048) [inline docs/en/connector-v2/sink/S3File.md]:
```suggestion
`CREATE_SCHEMA_WHEN_NOT_EXIST` ：Will Created when the path does not exist, use the path when the path is exist.        
```

File: docs/en/connector-v2/sink/S3File.md [hunk 3]
```diff
@@ -241,6 +246,22 @@ When File Format is Excel,The maximum number of data items that can be cached in
...
+Option introduction：  
+`RECREATE_SCHEMA` ：Will create when the path does not exist, delete and rebuild when the path is saved        
+`CREATE_SCHEMA_WHEN_NOT_EXIST` ：Will Created when the path does not exist, skipped when the path is saved        
+`ERROR_WHEN_SCHEMA_NOT_EXIST` ：Error will be reported when the path does not exist
+
+### data_save_mode[Enum]
+
+Before opening the synchronous task, the data file in the target path is differently processed.
+Option introduction：  
+`DROP_DATA`： Preserve path structure and delete data
```
Comment (EricJoy2048) [inline docs/en/connector-v2/sink/S3File.md]:
```suggestion
`DROP_DATA`： use the path but delete data files in the path.
```

File: docs/en/connector-v2/sink/S3File.md [hunk 4]
```diff
@@ -241,6 +246,22 @@ When File Format is Excel,The maximum number of data items that can be cached in
...
+`RECREATE_SCHEMA` ：Will create when the path does not exist, delete and rebuild when the path is saved        
+`CREATE_SCHEMA_WHEN_NOT_EXIST` ：Will Created when the path does not exist, skipped when the path is saved        
+`ERROR_WHEN_SCHEMA_NOT_EXIST` ：Error will be reported when the path does not exist
+
+### data_save_mode[Enum]
+
+Before opening the synchronous task, the data file in the target path is differently processed.
+Option introduction：  
+`DROP_DATA`： Preserve path structure and delete data  
+`APPEND_DATA`：Preserve path structure, preserve data
```
Comment (EricJoy2048) [inline docs/en/connector-v2/sink/S3File.md]:
```suggestion
`APPEND_DATA`：use the path, and add new files in the path for write datas.   
```

File: docs/en/connector-v2/sink/S3File.md [hunk 5]
```diff
@@ -241,6 +246,22 @@ When File Format is Excel,The maximum number of data items that can be cached in
...
+`CREATE_SCHEMA_WHEN_NOT_EXIST` ：Will Created when the path does not exist, skipped when the path is saved        
+`ERROR_WHEN_SCHEMA_NOT_EXIST` ：Error will be reported when the path does not exist
+
+### data_save_mode[Enum]
+
+Before opening the synchronous task, the data file in the target path is differently processed.
+Option introduction：  
+`DROP_DATA`： Preserve path structure and delete data  
+`APPEND_DATA`：Preserve path structure, preserve data   
+`ERROR_WHEN_DATA_EXISTS`：When there is data, an error is reported
```
Comment (EricJoy2048) [inline docs/en/connector-v2/sink/S3File.md]:
```suggestion
`ERROR_WHEN_DATA_EXISTS`：When there are some data files in the path, an error will reported.
```

File: docs/en/connector-v2/sink/S3File.md [hunk 6]
```diff
@@ -383,6 +404,8 @@ For orc file format simple config with `org.apache.hadoop.fs.s3a.SimpleAWSCreden
     access_key = "xxxxxxxxxxxxxxxxx"
     secret_key = "xxxxxxxxxxxxxxxxx"
     file_format_type = "orc"
+    schema_save_mode = "CREATE_SCHEMA_WHEN_NOT_EXIST"
```
Comment (EricJoy2048) [inline docs/en/connector-v2/sink/S3File.md:407]: Please add examples for `Multiple Table Writes`

File: docs/en/connector-v2/sink/S3File.md [hunk 7]
```diff
@@ -117,7 +117,12 @@ If write to `csv`, `text` file type, All column will be string.
...
 | hadoop_s3_properties             | map     | no       |                                                       | If you need to add a other option, you could add it here and refer to this [link](https://hadoop.apache.org/docs/stable/hadoop-aws/tools/hadoop-aws/index.html)       |
-|                                  |
+| schema_save_mode                 | Enum    | no       | CREATE_SCHEMA_WHEN_NOT_EXIST                          | Before turning on the synchronous task, do different treatment of the target path                                                                                     |
+| data_save_mode                   | Enum    | no       | APPEND_DATA                                           | Before opening the synchronous task, the data file in the target path is differently processed                                                                        |
+
+### path [string]
+
+Store the path of the data file to support variable replacement. For example: path=/test/${database_name}/${schema_name}/${table_name}
```
Comment (EricJoy2048) [inline docs/en/connector-v2/sink/S3File.md:125]: Please add examples for it.

File: docs/en/connector-v2/sink/S3File.md (hunk 6 above)
Comment (chl-wxp) [reply docs/en/connector-v2/sink/S3File.md:407]: added

File: docs/en/connector-v2/sink/S3File.md (hunk 7 above)
Comment (chl-wxp) [reply docs/en/connector-v2/sink/S3File.md:125]: added

File: docs/en/connector-v2/sink/S3File.md (hunk 1 above)
Comment (chl-wxp) [reply docs/en/connector-v2/sink/S3File.md]: updated

File: docs/en/connector-v2/sink/S3File.md (hunk 2 above)
Comment (chl-wxp) [reply docs/en/connector-v2/sink/S3File.md]: updated

File: docs/en/connector-v2/sink/S3File.md (hunk 3 above)
Comment (chl-wxp) [reply docs/en/connector-v2/sink/S3File.md]: updated

File: docs/en/connector-v2/sink/S3File.md (hunk 4 above)
Comment (chl-wxp) [reply docs/en/connector-v2/sink/S3File.md]: updated

File: docs/en/connector-v2/sink/S3File.md (hunk 5 above)
Comment (chl-wxp) [reply docs/en/connector-v2/sink/S3File.md]: updated

File: docs/en/connector-v2/sink/S3File.md [hunk 8]
```diff
@@ -383,10 +404,93 @@ For orc file format simple config with `org.apache.hadoop.fs.s3a.SimpleAWSCreden
...
+    username="xxxxxxxxxxxxx"
+    base-url="jdbc:mysql://localhost:3306/qa_source"
+    server-time-zone=UTC
+}
+}
+transform {
+}
+sink {
+S3File {
+    updateFieldType=null
```
Comment (EricJoy2048) [inline docs/en/connector-v2/sink/S3File.md]: It seems `updateFieldType`,`autoCreateShow `,`autoTable ` is not the options of S3 Sink Connector.
And `partition_dir_expression ` `is_partition_field_write_in_file ` are no need to config when `have_partition = false`.

Please check the examples config, It needs to be sufficiently correct.

File: docs/en/connector-v2/sink/S3File.md [hunk 9]
```diff
@@ -383,10 +404,79 @@ For orc file format simple config with `org.apache.hadoop.fs.s3a.SimpleAWSCreden
...
+Multi-table writing and saveMode
+
+```
+env {
+"job.name"="SeaTunnel_job"
+"job.mode"=STREAMING
+}
+source {
+MySQL-CDC {
+    updateFieldType=null
```
Comment (EricJoy2048) [inline docs/en/connector-v2/sink/S3File.md]:
```suggestion
    updateFieldType=null
    autoCreateShow=true
    autoCreate=false
    autoTable=false
```

File: docs/en/connector-v2/sink/S3File.md [hunk 10]
```diff
@@ -383,10 +404,82 @@ For orc file format simple config with `org.apache.hadoop.fs.s3a.SimpleAWSCreden
...
```
Comment (EricJoy2048) [inline docs/en/connector-v2/sink/S3File.md]:
```suggestion
    
```

File: docs/en/connector-v2/sink/S3File.md [hunk 11]
```diff
@@ -383,10 +404,82 @@ For orc file format simple config with `org.apache.hadoop.fs.s3a.SimpleAWSCreden
...
+    autoCreate=false
+    autoTable=false
+    autoCreateShow=true
+    autoCreate=false
+    autoTable=false
+    format=DEFAULT
+    "snapshot.split.size"=8096
+    "snapshot.fetch.size"=1024
+    "incremental.parallelism"=1
+    "connect.timeout.ms"=30000
```
Comment (EricJoy2048) [inline docs/en/connector-v2/sink/S3File.md]:
```suggestion
```

File: docs/en/connector-v2/sink/S3File.md [hunk 12]
```diff
@@ -383,10 +404,71 @@ For orc file format simple config with `org.apache.hadoop.fs.s3a.SimpleAWSCreden
...
+}
+source {
+MySQL-CDC {
+    
+    "connect.max-retries"=3
+    "connection.pool.size"=6
+    "chunk-key.even-distribution.factor.lower-bound"=0.05
+    "chunk-key.even-distribution.factor.upper-bound"=100
+    "sample-sharding.threshold"=1000
+    "inverse-sampling.rate"=1000
```
Comment (EricJoy2048) [inline docs/en/connector-v2/sink/S3File.md]:
```suggestion
```

File: seatunnel-common/src/main/java/org/apache/seatunnel/common/exception/CommonErrorCode.java [hunk 13]
```diff
@@ -24,6 +24,7 @@ public enum CommonErrorCode implements SeaTunnelErrorCode {
             "COMMON-02", "<identifier> JSON convert/parse '<payload>' operation failed."),
     UNSUPPORTED_DATA_TYPE(
             "COMMON-07", "'<identifier>' unsupported data type '<dataType>' of '<field>'"),
+    KERBEROS_AUTHORIZED_FAILED("COMMON-14", "Kerberos authorized failed"),
```
Comment (Hisoka-X) [inline seatunnel-common/src/main/java/org/apache/seatunnel/common/exception/CommonErrorCode.java]: Why add this?

Comment (chl-wxp) [reply seatunnel-common/src/main/java/org/apache/seatunnel/common/exception/CommonErrorCode.java]: deleted

**Comment by Hisoka-X** [comment]
image

File: seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/hadoop/HadoopFileSystemProxy.java [hunk 14]
```diff
@@ -140,6 +142,23 @@ public void createDir(@NonNull String filePath) throws IOException {
         }
     }
 
+    public List<LocatedFileStatus> fileList(String path) throws IOException {
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/hadoop/HadoopFileSystemProxy.java]:
```suggestion
    public List<LocatedFileStatus> listFile(String path) throws IOException {
```

Comment (Hisoka-X) [reply seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/hadoop/HadoopFileSystemProxy.java]: I prefer `listFile`

Comment (chl-wxp) [reply seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/hadoop/HadoopFileSystemProxy.java]: updated

**Review by Hisoka-X — APPROVED** [review]

**Review by EricJoy2048 — APPROVED** [review]

**Pull request merged**